next.config.js

```diff
@@ -1,14 +1,41 @@
+const glob = require('glob')
 const withPlugins = require('next-compose-plugins')
 const withCSS = require('@zeit/next-css')
 const withSourceMaps = require('@zeit/next-source-maps')
 const optimizedImages = require('next-optimized-images')
 const withOffline = require('next-offline')
+const webpack = require('webpack')
 
 const webpackExtra = require('./webpack.extra')
 
+require('dotenv').config()
+
 const nextConfiguration = {
-  webpack: config => Object.assign({}, config, webpackExtra),
+  webpack: (config, options) => {
+    const entryFactory = config.entry
+    const { isServer } = options
+    config.plugins.push(new webpack.EnvironmentPlugin(process.env))
```

Comment:
There is a problem here because you are adding the plugin just on server's webpack.

And I should avoid the use of `.push` function, instead you can use the `array destructor` to add the plugin.

Reply:
Could you tell me the another file that I need to include this config?

And about the array destructor the right way is like the code below?

```
config.plugins = [...config.plugins, new webpack.EnvironmentPlugin(process.env)]
```